src/avt/Pipeline/Data/avtDataRepresentation.C

```diff
@@ -98,22 +98,54 @@ ConvertVTKToVTKm(vtkDataSet *data)
 
         int dims[3];
         rgrid->GetDimensions(dims);
+        int nDims = (dims[2] == 1 ? 2 : 3);
```

Comment:
What happens if dims[0] or dims[1] is 1?

Reply:
I believe I've fixed this.

Reply:
I have no way to test this since VTKm doesn't work for 2D topology even in the case where dims[2] == 1. When it fails it results in an empty data set, resulting in a blank image, which isn't the worst failure mode. The logic for deciding if it is 2D is pretty conservative and only does it for the dims[2] == 1 case, leaving it in 3D for dims[0], dims[1] == 1. I have logic such that VisIt should never try to pass 2D data to a VTKm filter. Maybe one day VTKm will support 2D and I don't necessarily want to pull the code since I believe it is correct. Presumably anyone that pulls the tests for 2D would test that it works.